Changes for unstructured grids

fvitt opened the pull request:
This pull request includes:
 - removal of `igridh`, `xmet`, `ymet`, `dx`, `dy`, `xc`, `yc`
 - initialization of `igas` variables to zero
 - some code clean up

        modified:   calcrs.F90
        modified:   carma_globaer.h
        modified:   carma_types_mod.F90
        modified:   carmastate_mod.F90
        modified:   csolve.F90
        modified:   maxconc.F90
        modified:   nsubsteps.F90
        modified:   rhopart.F90
        modified:   setupatm.F90
        modified:   setupckern.F90
        modified:   setupgkern.F90
        modified:   setupgrow.F90
        modified:   setupvf_heymsfield2010.F90
        modified:   setupvf_std.F90
        modified:   setupvf_std_shape.F90
        modified:   sulfnucrate.F90
        modified:   supersat.F90
        modified:   vaporp_h2so4_ayers1980.F90
        modified:   vertical.F90

fvitt committed:
remove igridh, xmet, ymet, dx, dy, xc, yc; initial igas vars to zero; some code clean up

        modified:   calcrs.F90
        modified:   carma_globaer.h
        modified:   carma_types_mod.F90
        modified:   carmastate_mod.F90
        modified:   csolve.F90
        modified:   maxconc.F90
        modified:   nsubsteps.F90
        modified:   rhopart.F90
        modified:   setupatm.F90
        modified:   setupckern.F90
        modified:   setupgkern.F90
        modified:   setupgrow.F90
        modified:   setupvf_heymsfield2010.F90
        modified:   setupvf_std.F90
        modified:   setupvf_std_shape.F90
        modified:   sulfnucrate.F90
        modified:   supersat.F90
        modified:   vaporp_h2so4_ayers1980.F90
        modified:   vertical.F90

cbardeen commented on carma_globaer.h:
```diff
@@ -43,15 +43,9 @@
-#define ymet          cstate%f_ymet
 #define zmet          cstate%f_zmet
 #define zmetl         cstate%f_zmetl
-#define xc            cstate%f_xc
```
May want topreserve xc,yc,dx,dy for the stand-alone code. For example using it in an LES type environment were cartesian coordinates are more natural to identify a location than lat/lon.

cbardeen commented on carma_types_mod.F90:
```diff
@@ -400,9 +400,9 @@ module carma_types_mod
-    integer                                       :: f_igash2o
-    integer                                       :: f_igash2so4
-    integer                                       :: f_igasso2
+    integer                                       :: f_igash2o = 0
```
Why are these being initialized here? This doesn't happen for any other field. This is just meant to by the type definition. Initialization occurs within the object. In this case it is the carma object, so they should be initialized in carma_mod.F90. It current happens in CARMA_InitializeGrowth()

    ! Initialize indexes for the gases and check to make sure if H2SO4 is used
    ! that it occurs after H2O. This is necessary for supersaturation calculations.
    carma%f_igash2o   = 0
    carma%f_igash2so4 = 0
    carma%f_igasso2   = 0

I don't think these fields should be accessed unless do_growth is true, so if they are then we should double check that. Otherwise if they need to be initialized all the time, they could be set to 0 in CARMA_Initialize().

cbardeen commented on carma_types_mod.F90:
```diff
@@ -578,32 +578,18 @@ module carma_types_mod
-    !  ymet       Horizontal ds/dy (ds is metric distance)              {initatm}
     !  zmet       Vertical ds/dz (ds is metric distance)                {initatm}
     !  zmetl      Vertical ds/dz at edges (ds is metric distance)       {initatm}
-    !  xc         Horizontal position at center of box                  {initatm}
```
Again, may need to keep xc,yc,dx,dy

cbardeen commented on carmastate_mod.F90:
```diff
@@ -175,13 +161,6 @@ subroutine CARMASTATE_Create(cstate, carma_ptr, time, dtime, NZ, igridv, igridh,
     ! Calculate the metrics, ...
     ! if Cartesian coordinates were specifed, then the units need to be converted
     ! from MKS to CGS.
-    if (cstate%f_igridh == I_CART) then
```
Perhaps what we should do is get rid of lat/lon and make xc and yc either the center of the grid in lat/lon coordinates or in cartesian coordinates. I think the only place these are used are in error messages to indicate where the problem happened. So maybe they turn into optional fields and are indicates as arbitrary x and y locations only used for reporting? There are places in the model carma_model_mod.F90 for specific models wher ewe do use lat/lon. So perhaps we leave all 4, but make them all optional fields for CARMA, but always specify just the lat/lon in the CAM/CARMA case.

cbardeen commented on lusolvec_mod.F90:
```diff
@@ -70,10 +70,10 @@ module lusolvec_mod
   !! ******************************************************************  
   !! Version: 28.08.2000                                                 
   !! ******************************************************************
-  SUBROUTINE LUDCMPC(A,N,NP,INDX,D)
```
Okay to change it here. Does it collide with something? This isn't really a horizontal grid issue. Is this a problem with a certain compiler?

fvitt committed:
replace lat/lon coordinate variables the general x/y coordinate variables; remove gas indices init to 0

        modified:   calcrs.F90
        modified:   carma_globaer.h
        modified:   carma_types_mod.F90
        modified:   carmastate_mod.F90
        modified:   gsolve.F90
        modified:   microfast.F90
        modified:   tsolve.F90

fvitt commented:
All the stand-alone CARMA regression tests pass.

cbardeen commented:
Did they match the benchmarks too or just complete without error?

Thanks,

Chuck

> On Oct 11, 2022, at 8:46 AM, Francis Vitt ***@***.***> wrote:
> 
> 
> All the stand-alone CARMA regression tests pass.
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/ESCOMP/CARMA_base/pull/5#issuecomment-1274816716>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AFLRNRAA3N4E7OZDBRB2XY3WCV4SVANCNFSM55THN5TA>.
> You are receiving this because your review was requested.
>

fvitt commented:
> Did they match the benchmarks too or just complete without error? Thanks, Chuck

All of the benchmarks match.

cbardeen commented:
Great.

> On Oct 11, 2022, at 10:13 AM, Francis Vitt ***@***.***> wrote:
> 
> 
> Did they match the benchmarks too or just complete without error? Thanks, Chuck
> 
> All of the benchmarks match.
> 
> —
> Reply to this email directly, view it on GitHub <https://github.com/ESCOMP/CARMA_base/pull/5#issuecomment-1274946580>, or unsubscribe <https://github.com/notifications/unsubscribe-auth/AFLRNRESUHSJQD3UMGWMUH3WCWG33ANCNFSM55THN5TA>.
> You are receiving this because your review was requested.
>
